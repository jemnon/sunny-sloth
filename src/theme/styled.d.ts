import 'styled-components';
import {
  Breakpoints,
  Colors,
  ColorMode,
  FocusColors,
  Fonts,
  FontSizes,
  Header,
  HoverColors,
  MaxWidths,
  Spacing,
  ZIndex,
} from '../types/theme';

declare module 'styled-components' {
  export interface DefaultTheme {
    breakpoints: Breakpoints;
    colors: Colors;
    focusColors: FocusColors;
    fonts: Fonts;
    fontSizes: FontSizes;
    header: Header;
    hoverColors: HoverColors;
    maxWidths: MaxWidths;
    spacing: Spacing;
    transition: string;
    zIndex: ZIndex;
  }
}
