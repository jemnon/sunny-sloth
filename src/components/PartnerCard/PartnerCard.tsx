import React, { FC } from 'react';
import Img, { FluidObject } from 'gatsby-image';
import styled from 'styled-components';
import { Button } from '../Styles';
import Link from '../Link';

interface PartnerCardProps {
  description: {
    childMarkdownRemark: {
      html: string;
    };
  };
  image: { fluid: FluidObject };
  linkText?: string;
  linkUrl?: string;
  partnerName: string;
  promoCode?: string;
}

const PartnerCardArticle = styled.article`
  width: 100%;
`;

const PartnerCardFigure = styled.figure`
  margin-bottom: ${({ theme }): string => theme.spacing.m};
`;

const PartnerCardTitle = styled.h3`
  font-size: 1.3125rem;
  font-family: ${({ theme }): string => theme.fonts.latoBold};
  font-weight: 700;
  text-align: center;

  margin-bottom: ${({ theme }): string => theme.spacing.m};
`;

const PartnerCardDescription = styled.p`
  font-size: 1rem;
  text-align: center;

  &,
  & > p {
    margin-bottom: 0;
  }

  & + div {
    font-size: 1rem;
    margin-bottom: ${({ theme }): string => theme.spacing.m};
  }
`;

const PartnerCardPromoCode = styled.div`
  font-family: ${({ theme }): string => theme.fonts.latoBold};
  font-weight: 700;
  text-transform: uppercase;
  text-align: center;

  margin-bottom: ${({ theme }): string => theme.spacing.m};
`;

const PartnerCardContent = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

const PartnerCard: FC<PartnerCardProps> = ({
  description,
  image,
  linkText,
  linkUrl,
  partnerName,
  promoCode,
}) => {
  return (
    <PartnerCardArticle>
      <PartnerCardFigure>
        <Img alt={partnerName} fluid={image.fluid} />
      </PartnerCardFigure>
      <PartnerCardContent>
        <PartnerCardTitle>{partnerName}</PartnerCardTitle>
        <PartnerCardDescription
          dangerouslySetInnerHTML={{
            __html: description?.childMarkdownRemark?.html,
          }}
        />
        <PartnerCardPromoCode>{promoCode}</PartnerCardPromoCode>
        <Button>{linkText}</Button>
      </PartnerCardContent>
    </PartnerCardArticle>
  );
};

export default PartnerCard;
