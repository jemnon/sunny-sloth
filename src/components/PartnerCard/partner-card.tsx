import React, { FC } from 'react';
import Img, { FluidObject } from 'gatsby-image';
import styled from 'styled-components';
import { up } from 'styled-breakpoints';
import { ButtonLink } from '../Styles';

interface PartnerCardProps {
  description: {
    childMarkdownRemark: {
      html: string;
    };
  };
  image: { fluid: FluidObject };
  linkText?: string;
  linkUrl?: string;
  partnerName: string;
  promoCode?: string;
}

const PartnerCardArticle = styled.article`
  display: flex;
  flex-direction: column;
  width: 100%;
`;

const PartnerCardFigure = styled.figure`
  margin-bottom: ${({ theme }): string => theme.spacing.m};
`;

const PartnerCardTitle = styled.h3`
  font-size: 1.3125rem;
  font-family: ${({ theme }): string => theme.fonts.latoBold};
  font-weight: 700;
  text-align: center;

  margin-bottom: ${({ theme }): string => theme.spacing.m};
`;

const PartnerCardDescription = styled.p`
  font-size: 1rem;
  text-align: center;

  &,
  & > p {
    margin-bottom: 0;
  }

  & + div {
    font-size: 1rem;
    margin-bottom: ${({ theme }): string => theme.spacing.m};
  }

  & + button,
  & + a {
    margin-top: ${({ theme }): string => theme.spacing.m};
  }

  ${up('md')} {
    flex: 1;
    & + button,
    & + a {
      margin-top: 0;
    }
  }
`;

const PartnerCardPromoCode = styled.div`
  font-family: ${({ theme }): string => theme.fonts.latoBold};
  font-weight: 700;
  text-transform: uppercase;
  text-align: center;

  margin-bottom: ${({ theme }): string => theme.spacing.m};
`;

const PartnerCardContent = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  flex: 1;
`;

const PartnerCard: FC<PartnerCardProps> = ({
  description,
  image,
  linkText,
  linkUrl,
  partnerName,
  promoCode,
}) => {
  return (
    <PartnerCardArticle>
      <PartnerCardFigure>
        <Img alt={partnerName} fluid={image.fluid} />
      </PartnerCardFigure>
      <PartnerCardContent>
        <PartnerCardTitle>{partnerName}</PartnerCardTitle>
        <PartnerCardDescription
          dangerouslySetInnerHTML={{
            __html: description?.childMarkdownRemark?.html,
          }}
        />
        {promoCode && <PartnerCardPromoCode>{promoCode}</PartnerCardPromoCode>}
        <ButtonLink href={linkUrl} isInverted target="_blank">
          {linkText}
        </ButtonLink>
      </PartnerCardContent>
    </PartnerCardArticle>
  );
};

export default PartnerCard;
