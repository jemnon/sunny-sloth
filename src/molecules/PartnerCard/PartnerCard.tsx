import React, { FC, ReactNode } from 'react';
import styled from 'styled-components';
import Button from '../../atoms/Button';
import { H5 } from '../../atoms/Headings';
import ImgWrapper from '../../atoms/ImgWrapper';
import Paragraph from '../../atoms/Paragraph';
import Text from '../../atoms/Text';

interface PartnerCardProps {
  code: string;
  description?: string;
  image: ReactNode;
  onClick: (url: string) => void;
  title: string;
  url: string;
}

const PartnerCardContainer = styled.article`
  display: flex;
  align-items: center;
  flex-direction: column;
`;

const PartnerCardFigure = styled.figure`
  margin-bottom: ${({ theme: { spacing } }): string => spacing.sm2};
`;

const PartnerCard: FC<PartnerCardProps> = ({
  code,
  description,
  image,
  onClick,
  title,
  url,
}) => (
  <PartnerCardContainer aria-label={title}>
    <ImgWrapper as={PartnerCardFigure} ratio={9 / 16}>
      {image}
    </ImgWrapper>
    <H5>{title}</H5>
    {description && <Text textAlign="center">{description}:</Text>}
    <Text
      as={Paragraph}
      fontWeight="bold"
      fontSize="f1"
      textAlign="center"
      textTransform="uppercase"
    >
      {code}
    </Text>
    <Button variant="outline" onClick={(): void => onClick(url)}>
      Shop Now
    </Button>
  </PartnerCardContainer>
);

export default PartnerCard;
