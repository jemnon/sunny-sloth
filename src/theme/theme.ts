const theme = {
  breakpoints: {
    sm: '480px',
    md: '960px',
    lg: '1280px',
    xlg: '1440px',
  },
  fonts: {
    latoBold:
      `"Lato-Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    lato:
      `"Lato", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    latoItalic:
      `"Lato-Italic", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    noto:
      `"Noto Serif TC", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    notoBold:
      `"Noto Serif TC-Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
  },
  colors: {
    cream: '#f4ede6',
    darkTeal: '#162936',
    teal: '#244c53',
    lightBrown: '#d5a188',
    nearBlack: '#111',
    nearWhite: '#F9F9F9',
    darkOrange: '#BB5B34',
    orange: '#cf7651',
    white: '#ffffff',
    red: '#ff4136',
    green: '#19a974',
  },
  maxWidths: {
    sm: '480px',
    md: '960px',
    lg: '1280px',
  },
  spacing: {
    sm1: '0.25rem' /* 4px */,
    sm2: '0.5rem' /*  8px */,
    sm3: '0.75rem' /* 12px */,
    sm4: '1rem' /* 16px */,
    md1: '1.25rem' /* 20px */,
    md2: '1.5rem' /* 24px */,
    md3: '1.75rem' /* 28px */,
    md4: '2rem' /* 32px */,
    lg1: '2.25rem' /* 36px */,
    lg2: '2.5rem' /* 40px */,
    lg3: '2.75rem' /* 44px */,
    lg4: '3rem' /* 48px */,
  },
  transition: 'all 0.47s ease',
  zIndex: {
    z0: '0',
    z1: '1',
    z2: '2',
    z3: '3',
    z4: '4',
    z5: '5',
    z6: '6',
    z999: '999',
    z9999: '9999',
  },
};

export default theme;
