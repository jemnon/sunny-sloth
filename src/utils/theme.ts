import { DefaultTheme } from 'styled-components';

const theme: DefaultTheme = {
  breakpoints: {
    /* TODO: remove once conversion to styled-breakpoints is done */
    desktop: 'screen and (min-width: 960px)',
    tablet: 'screen and (min-width: 480px) and (max-width: 959px)',
    mobile: 'screen and (max-width: 479px)',
    /* ^^ */
    sm: '480px',
    md: '960px',
    lg: '1440px',
    xlg: '1920px',
  },
  fonts: {
    latoBold:
      `"Lato Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    lato:
      `"Lato", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    latoItalic:
      `"Lato Italic", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    noto:
      `"Noto Serif TC", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
  },
  colors: {
    cream: '#f4ede6',
    darkTeal: '#162936',
    teal: '#244c53',
    lightBrown: '#d5a188',
    nearBlack: '#111',
    nearWhite: '#F9F9F9',
    darkOrange: '#BB5B34',
    orange: '#cf7651',
    white: '#ffffff',
    red: '#ff4136',
    green: '#19a974',
  },
  maxWidths: {
    sm: '480px',
    md: '960px',
    lg: '1440px',
  },
  spacing: {
    s1: '0.25rem' /* 4px */,
    s2: '0.5rem' /*  8px */,
    s3: '0.75rem' /* 12px */,
    s4: '1rem' /* 16px */,
    s5: '1.25rem' /* 20px */,
    s6: '1.5rem' /* 24px */,
    s7: '1.75rem' /* 28px */,
    s8: '2rem' /* 32px */,
    s9: '2.25rem' /* 36px */,
    s10: '2.5rem' /* 40px */,
    s11: '2.75rem' /* 44px */,
    s12: '3rem' /* 48px */,
  },
  transition: 'all 0.47s ease',
  zIndex: {
    z0: '0',
    z1: '1',
    z2: '2',
    z3: '3',
    z4: '4',
    z5: '5',
    z6: '6',
    z999: '999',
    z9999: '9999',
  },
};

export default theme;
