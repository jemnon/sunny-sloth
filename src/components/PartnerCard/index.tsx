import PartnerCard from './partner-card';

export default PartnerCard;
