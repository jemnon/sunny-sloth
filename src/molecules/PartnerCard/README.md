# PartnerCard

### Description
Component description

```js
import PartnerCard from '@fenderdigital/ui/FILL IN TYPEPartnerCard';
```

### Props
* `propName` - property name and description 
