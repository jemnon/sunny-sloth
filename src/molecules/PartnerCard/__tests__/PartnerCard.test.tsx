import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import PartnerCard from '../PartnerCard';
import theme from '../../../theme';

describe('<PartnerCard />', (): void => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <PartnerCard
        code="WHISPEROFYUM"
        description="Some Desc"
        image={<img src="some-image.jpg" />}
        title="Title"
        url="/"
        onClick={jest.fn()}
      />
    </ThemeProvider>,
  );
  it('exists', (): void => {
    expect(PartnerCard).toBeDefined();
  });
  it('renders defaults correctly', (): void => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
