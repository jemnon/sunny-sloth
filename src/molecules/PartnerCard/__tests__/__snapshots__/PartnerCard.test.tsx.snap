// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PartnerCard /> renders defaults correctly 1`] = `
<article
  aria-label="Title"
  className="sc-jrAGrp cfzywa"
>
  <figure
    className="sc-kEjbxe leExdT"
  >
    <div
      className="sc-iBPRYJ bXkdcQ"
    >
      <img
        src="some-image.jpg"
      />
    </div>
  </figure>
  <h5
    className="sc-jSgupP kmFZoT"
  >
    Title
  </h5>
  <div
    className="sc-pFZIQ cZsunn"
  >
    Some Desc
    :
  </div>
  <p
    className="sc-fubCfw khejlI sc-pFZIQ gdjSXh"
    fontSize="f1"
    fontWeight="bold"
  >
    WHISPEROFYUM
  </p>
  <button
    className="sc-bdfBwQ cUBhvx"
    onClick={[Function]}
  >
    Shop Now
  </button>
</article>
`;
